server/grpc_service.go

```diff
@@ -1819,3 +1819,40 @@ func (s *GrpcServer) handleDamagedStore(stats *pdpb.StoreStats) error {
 	// TODO: reimplement add scheduler logic to avoid repeating the introduction HTTP requests inside `server/api`.
 	return s.GetHandler().AddEvictOrGrant(float64(stats.GetStoreId()), schedulers.EvictLeaderName)
 }
+
+// ReportMinResolvedTS implements gRPC PDServer.
+func (s *GrpcServer) ReportMinResolvedTS(ctx context.Context, request *pdpb.ReportMinResolvedTsRequest) (*pdpb.ReportMinResolvedTsResponse, error) {
+	forwardedHost := getForwardedHost(ctx)
+	if !s.isLocalRequest(forwardedHost) {
+		client, err := s.getDelegateClient(ctx, forwardedHost)
+		if err != nil {
+			return nil, err
+		}
+		ctx = grpcutil.ResetForwardContext(ctx)
+		return pdpb.NewPDClient(client).ReportMinResolvedTS(ctx, request)
+	}
+
+	if err := s.validateRequest(request.GetHeader()); err != nil {
+		return nil, err
+	}
+
+	rc := s.GetRaftCluster()
+	if rc == nil {
+		return &pdpb.ReportMinResolvedTsResponse{Header: s.notBootstrappedHeader()}, nil
+	}
+
+	var storage endpoint.MinResolvedTSStorage = s.storage
+	storeID := request.StoreId
+	minResolvedTS := request.MinResolvedTs
+
+	if err := storage.SaveMinResolvedTS(storeID, minResolvedTS); err != nil {
```

Comment:
If tikv sync resolved_ts every 1s, will it cause too many etcd writings and lead to performance issues?